packages/bitcore-lib/lib/script/interpreter.js

```diff
@@ -509,21 +661,230 @@ Interpreter.prototype.checkPubkeyEncoding = function(buf) {
   }
 
   // Only compressed keys are accepted in segwit
-  if ((this.flags & Interpreter.SCRIPT_VERIFY_WITNESS_PUBKEYTYPE) != 0 && this.sigversion == Interpreter.SIGVERSION_WITNESS_V0 && !PublicKey.fromBuffer(buf).compressed) {
+  if ((this.flags & Interpreter.SCRIPT_VERIFY_WITNESS_PUBKEYTYPE) != 0 && this.sigversion == Signature.Version.WITNESS_V0 && !PublicKey.fromBuffer(buf).compressed) {
     this.errstr = 'SCRIPT_ERR_WITNESS_PUBKEYTYPE';
     return false;
   }
 
   return true;
 };
 
+
+/**
+ * Verifies ECDSA signature
+ * @param {Signature} sig 
+ * @param {PublicKey} pubkey 
+ * @param {Number} nin 
+ * @param {Script} subscript 
+ * @param {Number} satoshis 
+ * @returns {Boolean}
+ */
+Interpreter.prototype.checkEcdsaSignature = function(sig, pubkey, nin, subscript, satoshis) {
+  var subscriptBuffer = subscript.toBuffer();
+  var scriptCodeWriter = new BufferWriter();
+  scriptCodeWriter.writeVarintNum(subscriptBuffer.length);
+  scriptCodeWriter.write(subscriptBuffer);
+
+  $.checkState(JSUtil.isNaturalNumber(satoshis));
+  var satoshisBuffer = new BufferWriter().writeUInt64LEBN(new BN(satoshis)).toBuffer();
+
+  var verified = SighashWitness.verify(
+    this,
+    sig,
+    pubkey,
+    nin,
+    scriptCodeWriter.toBuffer(),
+    satoshisBuffer
+  );
+  return verified;
+};
+
+
+/**
+ * Verifies Schnorr signature
+ * @param {Signature|Buffer} sig 
+ * @param {PublicKey|Buffer} pubkey 
+ * @param {Number} sigversion 
+ * @param {Object} execdata 
+ * @returns {Boolean}
+ */
+Interpreter.prototype.checkSchnorrSignature = function(sig, pubkey, sigversion, execdata) {
+  $.checkArgument(sig && Buffer.isBuffer(sig), 'Missing sig');
+  $.checkArgument(pubkey && Buffer.isBuffer(pubkey), 'Missing pubkey');
+  $.checkArgument(sigversion, 'Missing sigversion');
+  $.checkArgument(execdata, 'Missing execdata');
+
+  $.checkArgument(pubkey.length === 32, 'Schnorr signatures have 32-byte public keys. The caller is responsible for enforcing this.');
+  // Note that in Tapscript evaluation, empty signatures are treated specially (invalid signature that does not
+  // abort script execution). This is implemented in EvalChecksigTapscript, which won't invoke
+  // CheckSchnorrSignature in that case. In other contexts, they are invalid like every other signature with
+  // size different from 64 or 65.
+  if (sig.length !== 64 && sig.length !== 65) {
```

Comment:
`if (!(sig.length === 64 || sig.length === 65))` better expresses the spirit of the check